cnxepub/formatters.py

```diff
@@ -575,56 +576,27 @@ def _replace_exercises(elem, page_uuids):
                                        % (mathtext.encode('utf-8'), url))
 
         parent = elem.getparent()
-        if etree.QName(parent.tag).localname == 'p':
-            elem = parent
-            parent = elem.getparent()
-
-        parent.remove(elem)  # Special case - assumes single wrapper elem
         for child in nodes:
-            parent.append(child)
+            parent.insert(parent.index(elem), child)
+        parent.remove(elem)  # Special case - assumes single wrapper elem
 
     xpath = '//xhtml:a[contains(@href, "{}")]'.format(match)
```

Comment:
@phil We worked on changing this line (xpath selection & replacement) to fit with the new cnxml exercise format. But it turns out that, with the above changes to how new elements are injected, the same selector can still be used. What would be best: trying to come up with a way to change the selector (this was the bug we were still stuck on when we paired the first time), or keeping it as is?

Reply:
Oh, I forgot about this one. I _think_ keeping this here will work but I'm not 100% sure. If you're able to use this code to build multiple books we can verify this code still builds the other books that uses embedded exercises (I think it's microbiology)